### DISCUSSION TITLE
Ability to Customize Filename Format for Camera Scans or New Uploads

### DISCUSSION THREAD

**Discussion by fire1ce** [LeoKlaus/Paperparrot#148] [Feature Requests]
I would like to request the addition of a feature that allows users to configure a predefined filename format for documents added through camera scans and the uploads. This feature would enhance the organization of files and streamline the saving process, making it more efficient and user-friendly.

Example Use Case:
Suppose a user wishes to save scanned documents in a structured directory format, based on the date of the scan and a customizable portion of the filename. They could define a filename format such as:

```text
{year}/{month}/{day}_customText
```

In this format:

{year}, {month}, and {day} would be placeholders that automatically get replaced with the current year, month, and day, respectively.
customText is a user-defined text string that remains constant or could be manually altered for each scan.
This feature would provide users with the flexibility to maintain a consistent and organized filing system, catering to various use cases and preferences.

PS: I wanted to take a moment to express my appreciation for the work put into this app. It looks very promising, and I am excited about its potential. Keep up the great work!

**Comment by LeoKlaus** [comment]
Hi fire1ice, thanks for reaching out and your kind words!

I'm currently working on something very similar for the export of files, this shouldn't be too hard to do.

**Reply by fire1ce** [reply]
Great. I will open some features or enchantments  requests. Don’t be overwhelmed. Your app is the most promising from the alternatives so i would like to suggest the improvements.

**Comment by LeoKlaus** [comment]
Added in 1.9.0
